flappy-bird.js

```diff
@@ -198,18 +207,43 @@ export class Bird extends Scene {
         for (let i = 0; i < this.pipe_num; i++) {
             const pipe_len = this.pipe_lens[i];
 
-            //draw the top pipes
+            //draw the bottom pipes
             const bottom_pipe_model_transform = model_transform.times(Mat4.translation(0, pipe_len - 11, i * this.pipe_distance))
             this.draw_pipe(context, program_state, bottom_pipe_model_transform, pipe_len);
 
-            //draw bottom pipe
+            //draw top pipe
             const top_pipe_model_transform = model_transform.times(Mat4.translation(0, this.pipe_gap - (9 - pipe_len), i * this.pipe_distance))
-                .times(Mat4.rotation(Math.PI, 1, 0, 0));
+                                                            .times(Mat4.rotation(Math.PI, 1, 0, 0));
             this.draw_pipe(context, program_state, top_pipe_model_transform, 9 - pipe_len);
+
+            //determine collision on top and bottom
```

Comment:
The whole logic can be abstract to a function, say `checkCollision(top_pipe, bottom_pipe)`